packages/std/src/results/cosmos_msg.rs

```diff
@@ -22,56 +22,72 @@ where
     Wasm(WasmMsg),
 }
 
+/// The message types of the bank module.
+///
+/// See https://github.com/cosmos/cosmos-sdk/blob/v0.40.0/proto/cosmos/bank/v1beta1/tx.proto
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
 #[serde(rename_all = "snake_case")]
 pub enum BankMsg {
-    // this moves tokens in the underlying sdk
+    /// Sends native tokens from the contract to the given address.
+    ///
+    /// This is translated to a [MsgSend](https://github.com/cosmos/cosmos-sdk/blob/v0.40.0/proto/cosmos/bank/v1beta1/tx.proto#L19-L28).
+    /// `from_address` is automatically filled with the current contract's address.
     Send {
-        from_address: HumanAddr,
         to_address: HumanAddr,
         amount: Vec<Coin>,
     },
 }
 
+/// The message types of the staking module.
+///
+/// See https://github.com/cosmos/cosmos-sdk/blob/v0.40.0/proto/cosmos/staking/v1beta1/tx.proto
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
 #[serde(rename_all = "snake_case")]
 pub enum StakingMsg {
-    Delegate {
-        // delegator is automatically set to address of the calling contract
-        validator: HumanAddr,
-        amount: Coin,
-    },
-    Undelegate {
-        // delegator is automatically set to address of the calling contract
-        validator: HumanAddr,
-        amount: Coin,
-    },
+    /// This is translated to a [MsgDelegate](https://github.com/cosmos/cosmos-sdk/blob/v0.40.0/proto/cosmos/staking/v1beta1/tx.proto#L81-L90).
```

Comment:
Nice set of comments (down the line).
Will save future questions and people can look at the sdk code to understand what is returned in certain conditions (eg. 3 separate delegations)